File: gemini_tests/python/command_injection.py
import os
import subprocess

# Simulate user input from a web request or CLI
user_input = "example.com; echo 'pwned by command injection'"

# --- VULNERABLE CODE (os.system) ---
# Using os.system with untrusted input
command = f"ping -c 1 {user_input}"
os.system(command) # CWE-78: Improper Neutralization of Special Elements used in an OS Command ('OS Command Injection')
# ------------------------------------

# --- VULNERABLE CODE (subprocess) ---
# Using subprocess with shell=True
subprocess.run(command, shell=True) # CWE-78
# ------------------------------------

File: gemini_tests/javascript/command_injection.js
const { exec } = require('child_process');

// Simulate user input from a web request
const userInput = "8.8.8.8; echo 'pwned by command injection'";

// --- VULNERABLE CODE ---
// Using exec with untrusted input is dangerous
const command = `ping -c 1 ${userInput}`;
exec(command, (error, stdout, stderr) => { // CWE-78: Improper Neutralization of Special Elements used in an OS Command ('OS Command Injection')
  if (error) {
    console.error(`exec error: ${error}`);
    return;
  }
  console.log(`stdout: ${stdout}`);
  console.error(`stderr: ${stderr}`);
});
// -----------------------